### PR TITLE
[MU4] Fix #9902: Can't adjust line or slur length via keyboard

### LINKED ISSUE #9902: [MU4 Issue] Can't adjust line or slur length via keyboard
It is not possible to make lines longer or shorter via the keyboard. Affects "spanner" elements with draggable handles:

- hairpins _(also unable to adjust height of "jaws")_
- slurs & ties _(also unable to adjust curvature)_
- cresc./dim.
- ottavas (8va/8vb)
- pedal markings
- trill lines
- etc.

__Expected behaviour__

1. On **entering** (i.e. adding/inserting) one of the above elements, it should become selected and the **rightmost** handle should gain keyboard focus.

image

2. On **navigating** to one of the above elements with Alt+Left/Alt+Right, the **leftmost** handle should gain keyboard focus.

image

3. Pressing the Tab key should move focus clockwise to one of the other handles (Shift+Tab to go anticlockwise). But see suggested change below.

**Actual behaviour**

1. Line elements are not selected when they are entered in the score.
2. The drag handles do not gain keyboard focus during navigation with Alt+Left and Alt+Right (you must press Alt+Shift+E to edit the line).
3. The Tab shortcut moves focus away from the score instead of moving to a different handle. This is sometimes desirable, sometimes not. See suggestion below.

**Suggested change to Tab navigation**

MuseScore 3 used the Tab key to navigate between drag handles in a closed loop, which prevented Tab being used to navigate to other UI elements like the Inspector (see bug [MU#299941](https://musescore.org/en/node/299941)). In MuseScore 4, perhaps we should "break the loop"? This means:

- Pressing **Tab** would still move you to the **next handle (right)**, but when you reach the **rightmost handle**, pressing Tab again would navigate to a UI panel on the **right** of the score (or below it, like the status bar).
- Pressing **Shift+Tab** would still move you to the **previous handle (left)**, but when you reach the **leftmost handle**, pressing Shift+Tab again would navigate to a UI panel on the **left** of the score (e.g. the Palettes).

Basically, each handle would be treated as a separate UI element for Tab and Shift+Tab, and when you run out of handles you then you start to navigate the rest of the UI.

### PR DESCRIPTION
**Pull request by RomanPudashkin**
Resolves: 
https://github.com/musescore/MuseScore/issues/10209
https://github.com/musescore/MuseScore/issues/9902

### CODE REVIEW

File: src/engraving/libmscore/edit.cpp [hunk 1]
```diff
@@ -1959,7 +1959,7 @@ void Score::addHairpin(HairpinType type)
         for (int staffIdx = selection().staffStart(); staffIdx < selection().staffEnd(); ++staffIdx) {
             ChordRest* cr1 = selection().firstChordRest(staffIdx * VOICES);
             ChordRest* cr2 = selection().lastChordRest(staffIdx * VOICES);
-            addHairpin(type, cr1, cr2, /* toCr2End */ true);
+            return addHairpin(type, cr1, cr2, /* toCr2End */ true);
```
Comment (Jojo-Schmitz) [inline src/engraving/libmscore/edit.cpp:1962]: Wouldn't that leave the function in the first round of that `for` loop? The comment says `// add hairpin on each staff if possible`

Comment (RomanPudashkin) [reply src/engraving/libmscore/edit.cpp:1962]: Thanks, I didn't notice this for loop :(

Comment (RomanPudashkin) [reply src/engraving/libmscore/edit.cpp]: fixed

**Comment by DmitryArefiev** [comment]
Tested #10209 #9902 on Win10, Mac12, LinuxMint 20.2 - FIXED      

Please merge into master

**Review by vpereverzev — APPROVED** [review]

**Pull request merged**